server/views/serviceProviderReferrals/editSession.njk

```diff
@@ -12,21 +12,31 @@
 {% block pageContent %}
   <div class="govuk-grid-row">
     <div class="govuk-grid-column-two-thirds">
-      <form method="post" novalidate>
-        <input type="hidden" name="_csrf" value="{{csrfToken}}">
+      {% if errorSummaryArgs !== null %}
+        {{ govukErrorSummary(errorSummaryArgs) }}
+      {% endif %}
+
+      <form method="post" novalidate="novalidate">
 
-        {% if errorSummaryArgs !== null %}
-          {{ govukErrorSummary(errorSummaryArgs) }}
-        {% endif %}
+        <input type="hidden" name="_csrf" value="{{csrfToken}}">
 
         <h1 class="govuk-heading-xl">{{ presenter.text.title }}</h1>
 
-        {{ govukDateInput(dateInputArgs) }}
-        {{ appTimeInput(timeInputArgs) }}
-        {{ govukDateInput(durationDateInputArgs) }}
+        <div id='session-input' class="{{ serverError.classes }}">
+          {% if serverError.text !== null %}
```

Comment:
Perhaps could encapsulate it into a single error object?

{ message: string; error: {field: string; message: string}[], classes: string }

and have serverError() return this or `null` rather than checking `text`?

Reply:
Good plan, I've done this, although without the `error` field inside.